simulator/modules/core/perf_instr.h

```diff
@@ -51,6 +51,9 @@ class PerfInstr : public FuncInstr
                                                 this->is_store() ||
                                                 this->is_jump()  ||
```

Comment:
Probably, you have to exclude `this->is_jump()` from here.

Reply:
It seems to me that it is better to remove ``this->is_indirect_jump()`` from ``is_branch_stage_required()``?

Reply:
I don't think so. First of all, it just contradict with common sense — branch stage is required for branches, and not required for anything else. Additionally, MIPS has "branch-and-link" instructions, which store return address likewise `jal`